solr/test-framework/src/java/org/apache/solr/SolrTestCaseJ4.java

```diff
@@ -3235,6 +3236,14 @@ private static boolean isChildDoc(Object o) {
     return o instanceof SolrInputDocument;
   }
 
+  public static String getUrlFrom(SolrClient solrClient) {
+    if (solrClient instanceof HttpSolrClient) {
+      return ((HttpSolrClient) solrClient).getBaseURL();
+    } else {
+      return null;
```

Comment:
I *think* that this should raise an exception, not a null...   since it's test code, we wouldn't ever call this where a null would make sense?   I *guess* maybe if you had random types of `SolrClient` being created you would need an else?   Right now however that owuldn't happen.

Reply:
> I _guess_ maybe if you had random types of `SolrClient` being created you would need an else?

If its ok to add `getBaseURL()` method to other clients, I intend to use else if statements to check `instanceOf `other clients and the final `else `statement to raise an exception